crates/codegen/utils/src/internal/formatting.rs

```diff
@@ -74,7 +74,7 @@ fn generate_header(file_path: &PathBuf) -> Result<String> {
     return match get_extension(file_path)? {
         "json" => Ok(format!("")),
         "html" | "md" => Ok(format!("<!-- {warning_line} -->")),
-        "rs" => Ok(format!("// {warning_line}")),
+        "rs" => Ok(format!("// {warning_line}\n#![allow(clippy::all)]")),
```

Comment:
I did that as part of an incremental improvement as Clippy is not configured yet, but I imagine we might want to enable it soon-ish.

> I imagine we would also run it over generated files (...)

While this makes sense in principle, sometimes it's hard to do that depending on how we emit the generated code and adapting the codegen code to always emit a Clippy-happy code might be more work than it's worth. Short of reformatting the code, we don't really optimize the generated code for legibility right now AIUI, so we probably wouldn't spend much resources to also appease the linter as well.

However, we can definitely pick some low-hanging fruits like redundant copies or unneeded refs when possible, once we enable the linter, and re-evaluate from there.

I can remove the change so that we can revisit it once we explore integrating Clippy.

Reply:
That sounds like a good starting point, I've reverted the blanket `allow(clippy::all)` for now.